src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs

```diff
@@ -128,6 +129,8 @@ private void RunOneRoundOfInitialization(CancellationToken cancellationToken)
 
                 if (step.MustInitialize)
                 {
+                    if (task.IsFaulted && task.Exception is not null)
+                        ExceptionDispatchInfo.Capture(task.Exception.GetBaseException()).Throw();
```

Comment:
But what if the exception is not yet thrown?
I am not convinced this should be here.

Reply:
The problem is not when exception hasn't been thrown yet as it will be handled when task is awaited. However, when it gets thrown before `Task.WhenAll(_allPending)` it will not break initialization process. `ExecuteStep' will mark the stage as `Complete` regardless of exception and next steps will continue execution.

Reply:
Reworked this check to be called in `InitializeAll`. Tasks which have to be initialized but fail due to an exception will be marked as `Failed`, to stop execution of steps that depend on them.